polynote-server/src/main/scala/polynote/server/Server.scala

```diff
@@ -169,8 +169,18 @@ class Server {
             .handleSome(serveStatic)
             .build
         }
+      
+      def initNotebookStorageDir() = {
+        val NotebookStorageDirPath = currentPath.resolve(config.storage.dir)
+        if (NotebookStorageDirPath.toFile.exists()) {
```

Comment:
`exists()` is a side effect here, so it needs to be in an `effectBlocking` of its own.

Also, should use `Files.createDirectories` in case multiple parent directories need to be created.

Actually, it looks like `createDirectories` won't cause a failure if the directory already exists, so the `exists()` check might not even be necessary when using that method.

Reply:
Thank you for your advice, I modified again and passed the manual test.